Automattic/newspack-blocks
# Inline HPB styles in a style tag

### All Submissions:






### Changes proposed in this Pull Request:



Inlines the critical parts of CSS needed by the Homepage Posts Block, in order to get rid of [layout shift](https://web.dev/cls/) happening when the file-based CSS stylesheet is loaded. 

To demonstrate, here's a recording of a (throttled) page load on `master`:

<img src="https://github.com/Automattic/newspack-blocks/assets/7383192/49a0645e-cbec-488a-8887-c1c751d1231e" width="300px">

And here's how it looks like on this branch:

<img src="https://github.com/Automattic/newspack-blocks/assets/7383192/d16eebde-75b5-4bc9-88f3-083c74113e5c" width="300px">

If this strategy proves effective in production environment, it can be enhanced by moving more of the static styles from `view.css` to the dynamically created inline stylesheet. 

### How to test the changes in this Pull Request:

1. Create a page with some instances of the Homepage Posts Block, with different type scale settings and images shown
2. Load the front-end, observe the styles are applied just like on `master`, only faster*

\* use DevTools' Network speed throttling to observe it better

### Other information:
